pandas/tests/types/test_dtypes.py

```diff
@@ -206,6 +208,162 @@ def test_parser(self):
                 DatetimeTZDtype('ns', tz),
             )
 
+    def test_empty(self):
+        dt = DatetimeTZDtype()
+        with tm.assertRaises(AttributeError):
+            str(dt)
+
+
+class TestPeriodDtype(Base, tm.TestCase):
+
+    def setUp(self):
+        self.dtype = PeriodDtype('D')
+
+    def test_construction(self):
+        with tm.assertRaises(ValueError):
+            PeriodDtype('xx')
+
+        for s in ['period[D]', 'Period[D]', 'D']:
+            dt = PeriodDtype(s)
+            self.assertEqual(dt.freq, pd.tseries.offsets.Day())
+            self.assertTrue(is_period_dtype(dt))
+
+        for s in ['period[3D]', 'Period[3D]', '3D']:
+            dt = PeriodDtype(s)
+            self.assertEqual(dt.freq, pd.tseries.offsets.Day(3))
+            self.assertTrue(is_period_dtype(dt))
+
+        for s in ['period[26H]', 'Period[26H]', '26H',
+                  'period[1D2H]', 'Period[1D2H]', '1D2H']:
+            dt = PeriodDtype(s)
+            self.assertEqual(dt.freq, pd.tseries.offsets.Hour(26))
+            self.assertTrue(is_period_dtype(dt))
+
+    def test_subclass(self):
+        a = PeriodDtype('period[D]')
+        b = PeriodDtype('period[3D]')
+
+        self.assertTrue(issubclass(type(a), type(a)))
+        self.assertTrue(issubclass(type(a), type(b)))
+
+    def test_identity(self):
+        self.assertEqual(PeriodDtype('period[D]'),
+                         PeriodDtype('period[D]'))
+        self.assertIs(PeriodDtype('period[D]'),
+                      PeriodDtype('period[D]'))
+
+        self.assertEqual(PeriodDtype('period[3D]'),
+                         PeriodDtype('period[3D]'))
+        self.assertIs(PeriodDtype('period[3D]'),
+                      PeriodDtype('period[3D]'))
+
+        self.assertEqual(PeriodDtype('period[1S1U]'),
+                         PeriodDtype('period[1000001U]'))
+        self.assertIs(PeriodDtype('period[1S1U]'),
+                      PeriodDtype('period[1000001U]'))
+
+    def test_coerce_to_dtype(self):
+        self.assertEqual(_coerce_to_dtype('period[D]'),
+                         PeriodDtype('period[D]'))
+        self.assertEqual(_coerce_to_dtype('period[3M]'),
+                         PeriodDtype('period[3M]'))
+
+    def test_pandas_dtype(self):
+        self.assertEqual(pandas_dtype('period[D]'),
+                         PeriodDtype('period[D]'))
+        self.assertEqual(pandas_dtype('period[3M]'),
```

Comment:
maybe some of these tests (in PeriodDtype) could be moved higher up to more generally test DatetimewithTZ ?